src/briefcase/platforms/web/static.py

```diff
@@ -0,0 +1,347 @@
+import subprocess
+import sys
+import webbrowser
+from http.server import HTTPServer, SimpleHTTPRequestHandler
+from pathlib import Path
+from zipfile import ZipFile
+
+from briefcase.commands import (
+    BuildCommand,
+    CreateCommand,
+    OpenCommand,
+    PackageCommand,
+    PublishCommand,
+    RunCommand,
+    UpdateCommand,
+)
+from briefcase.config import AppConfig
+from briefcase.exceptions import BriefcaseCommandError
+
+
+class StaticWebMixin:
+    output_format = "static"
+    platform = "web"
+
+    def project_path(self, app):
+        return self.bundle_path(app) / "www"
+
+    def binary_path(self, app):
+        return self.bundle_path(app) / "www" / "index.html"
+
+    def wheel_path(self, app):
+        return self.project_path(app) / "static" / "wheels"
+
+    def distribution_path(self, app, packaging_format):
+        return self.platform_path / f"{app.formal_name}-{app.version}.zip"
+
+
+class StaticWebCreateCommand(StaticWebMixin, CreateCommand):
+    description = "Create and populate a static web project."
+
+    def output_format_template_context(self, app: AppConfig):
+        """Add style framework details to the app template."""
+        return {
+            "style_framework": getattr(app, "style_framework", "None"),
+        }
+
+
+class StaticWebUpdateCommand(StaticWebCreateCommand, UpdateCommand):
+    description = "Update an existing static web project."
+
+
+class StaticWebOpenCommand(StaticWebMixin, OpenCommand):
+    description = "Open an existing static web project."
+
+
+class StaticWebBuildCommand(StaticWebMixin, BuildCommand):
+    description = "Build a static web project."
+
+    def _trim_file(self, path, sentinel):
+        """Re-write a file to strip any content after a sentinel line.
+
+        The file is stored in-memory, so it shouldn't be used on files
+        with a *lot* of content before the sentinel.
+
+        :param path: The path to the file to be trimmed
+        :param sentinel: The content of the sentinel line. This will
+            become the last line in the trimmed file.
+        """
+        content = []
+        with path.open("r", encoding="utf-8") as f:
+            for line in f:
+                if line.rstrip("\n") == sentinel:
```

Comment:
Unless something really interesting happens with file iteration, I would expect `rstrip('\n')` to ever actually find a line break to strip.

Reply:
`for line in f` includes the `\n` on the line being read, so it should match on *every* line.